# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [91]:
import pandas as pd

try:
    data=pd.read_csv("C:\\Users\Ahsas\Documents\ЯНДЕКС_ПРАКТИКУМ\pdf_работы\data\data _2.csv")
except:
    data=pd.read_csv('/datasets/data.csv')

# прочитайте csv-файл

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [92]:
data.head(21)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   
20         1   -1311.604166         36              среднее             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   
20   женат / замужем                 0      M   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свад

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [93]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [94]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [95]:
data['income_type'].value_counts()

income_type
сотрудник          11119
компаньон           5085
пенсионер           3856
госслужащий         1459
безработный            2
предприниматель        2
студент                1
в декрете              1
Name: count, dtype: int64

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [96]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [97]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

In [98]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [99]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5], dtype=int64)

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [100]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [101]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [102]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [103]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [104]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [105]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [106]:
data.duplicated().sum()

71

In [107]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [108]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [109]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [110]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [111]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [112]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)
data.head()

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose total_income_category          purpose_category  
0               покупка жилья                     B  операции с недвижимостью  
1     приобретение автомобиля                     C    операции с автомобилем  
2               покупка жилья                     C  операции с недвижимостью  
3  дополнительное образование                     B     получение образования  
4             сыграть свадьбу                     C        проведение свадьбы

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [113]:
data.head()

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose total_income_category          purpose_category  
0               покупка жилья                     B  операции с недвижимостью  
1     приобретение автомобиля                     C    операции с автомобилем  
2               покупка жилья                     C  операции с недвижимостью  
3  дополнительное образование                     B     получение образования  
4             сыграть свадьбу                     C        проведение свадьбы

In [114]:
q_children = data.pivot_table(index = [ 'children'], values = 'debt', aggfunc = {'count','sum'})
q_children = q_children.rename(columns={'count':'no_debts','sum':'no_debt_overdue'})
#Здесь и далее переменные:
#no_debts- число случаев задолженности по кредиту
#share_overdue - % неплатежей\задолженности по кредиту в категории
#% - доля категории заемщика от общего числа

q_children['share_overdue'] =100*round(q_children['no_debt_overdue']/q_children['no_debts'],3)
q_children=q_children.sort_values(by='share_overdue', ascending=False)
#расчет доли группы заемщиков от общего числа заемщиков
q_children['%_of_total_debts']=100*round(q_children['no_debts']/q_children['no_debts'].sum(),4)
q_children

no_debts  no_debt_overdue  share_overdue  %_of_total_debts
children                                                            
4               41                4            9.8              0.19
2             2052              194            9.5              9.62
1             4808              444            9.2             22.54
3              330               27            8.2              1.55
0            14091             1063            7.5             66.06
5                9                0            0.0              0.04

**Вывод:** 
Выявлены незначительные девиации в уровне просрочки платежей по кредиту семей с разным числом детей. 

    - заемщики без детей - самая многочисленная группа и имеют самый низкий (7.5%) процент просрочки, за исключением семей с         пятью детьми (0%), число которых в выборке незначительное, лишь 0.04%.
    - доля просрочек на 0.7% возрастает в сегменте заемщиков с двумя детьми, по сравнению с семьями с одним                           ребенком.
    - в то время, как семьи с тремя детьми допускают просрочку несколько реже -8.2%, чем с двумя. Предположительно, достигая         большего уровня достатка ко времени рождения третьего ребенка.
    - семьи с четырьмя детьми имеют наивысший уровень просрочки из всех сегментов выборки.
   Таким образом, по мере роста числа детей в семье, риск просрочки возрастает, а доля просрочки выплат возрастает с шагом 0.7-    1%, на примере исследованных данных.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [115]:
q_family_st = data.pivot_table(index = [ 'family_status'], values = 'debt', aggfunc = {'count','sum'})
q_family_st = q_family_st.rename(columns={'count':'no_debts','sum':'no_debt_overdue'})
q_family_st['share_overdue'] =100*round(q_family_st['no_debt_overdue']/q_family_st['no_debts'],3)
q_family_st=q_family_st.sort_values(by='share_overdue', ascending=False)

#расчет доли группы заемщиков от общего числа заемщиков
q_family_st['%']=100*round(q_family_st['no_debts']/q_family_st['no_debts'].sum(),4)

q_family_st

no_debts  no_debt_overdue  share_overdue      %
family_status                                                         
Не женат / не замужем      2796              273            9.8  13.11
гражданский брак           4134              385            9.3  19.38
женат / замужем           12261              927            7.6  57.48
в разводе                  1189               84            7.1   5.57
вдовец / вдова              951               63            6.6   4.46

**Вывод:**
    ** Заемщики, не состоявшие в браке, имеют наибольший процент просрочки - 0.098%. Сраввнимые по уровню - группа заемщиков,          состоящих в гражданском браке. Предположительно, и те и другие - самая молодая страта в выборке, обладают меньшим              уровнем финансовой стабильности.
    ** заемщики, состоящие в браке или в разводе незначительно лучше по уровню просрочки - 0.076% и 0.071% соответственно.
    ** вдовцы демонстрируют самый низкий процент просрочки. Вероятно, в силу педантичности и стабильного достатка                      представителей       старшего поколения.


#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [116]:
q_income_level = data.pivot_table(index = [ 'total_income_category'], values = 'debt', aggfunc = {'count','sum'})
q_income_level = q_income_level.rename(columns={'count':'no_debts','sum':'no_debt_overdue'})
q_income_level['share_overdue'] =100*round(q_income_level['no_debt_overdue']/q_income_level['no_debts'],3)
#расчет доли группы заемщиков от общего числа заемщиков
q_income_level['%']=100*round(q_income_level['no_debts']/q_income_level['no_debts'].sum(),4)

q_income_level=q_income_level.sort_values(by='share_overdue', ascending=False)
q_income_level

no_debts  no_debt_overdue  share_overdue      %
total_income_category                                                 
E                            22                2            9.1   0.10
C                         15921             1353            8.5  74.64
A                            25                2            8.0   0.12
B                          5014              354            7.1  23.51
D                           349               21            6.0   1.64

**Вывод:** 
Наибольший уровень просрочки в распределении по уровню дохода демонстрирует самая многочисленная группа по уровню достатка - группа С- 8.5% (кроме группы E, числом которых можно пренебречь, т.к. доля таких заемщиков составляет только 0.1%).
Группа В - более обеспеченных заемщиков допускает просрочку на 1.4% случаев реже, доля кредитов заемщиков категории B ниже в 3 раза по сравнению с группой С.
Самые малообеспеченные -D, опуская группу Е, занимают самую низкую доллю (6%) просроченной задолженности.
Самые богатые не отличаются дисциплиной со своими 8% просрочки...



Извините, я не понимаю, что вы имеете в виду под "несблансированностью выборок". Мне казалось я в выводе как раз сделала акцент на две самые многочисленные группы. Нет?
Изменила порядок сортировки выборки. Теперь по убыванию по параметру доли просрочки.
Или код подправить, чтобы самые малочисленные в группу Other,например, были объеденены? Поясните, спасибо.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [117]:
q_purpose = data.pivot_table(index = ['purpose_category'], values = 'debt', aggfunc = {'count','sum'})
q_purpose = q_purpose.rename(columns={'count':'no_debts','sum':'no_debt_overdue'})
q_purpose['share_overdue']=100*round(q_purpose['no_debt_overdue']/q_purpose['no_debts'],3)
#расчет доли группы заемщиков от общего числа заемщиков
q_purpose['%']=100*round(q_purpose['no_debts']/q_purpose['no_debts'].sum(),4)

q_purpose=q_purpose.sort_values(by='purpose_category', ascending=True)
q_purpose



no_debts  no_debt_overdue  share_overdue      %
purpose_category                                                         
операции с автомобилем        4279              400            9.3  20.06
операции с недвижимостью     10751              780            7.3  50.40
получение образования         3988              369            9.3  18.70
проведение свадьбы            2313              183            7.9  10.84

**Вывод:** 
заемщики для целей оплаты образования и покупки автомобиля имеют одинаково высокую долю просрочки - 9.3%.
Две прочие группы - операции с недвижимостью и проведение свадьбы - значительно ниже - 7.3% и 7.9% соответственно.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* Т.к. число пропущенных значений в двух параметрах - общий трудовой стаж и ежемесячный доход - одинаковое, возможно 2174 кредита оформили клиенты, не являющиеся "зарплатными" (банк-кредитор не предоставляет услуги РКО организации-работодателя потенциального заемщика) для кредитора. Т.о. кредитор не получает актуальной информации об уровне дохода и стаже заемщика.

не понимаете, что значит "зарплатный клиент"? это как? m
Никаких зависимостей здесь я не вижу. Наверное можно было не заполнять эти данные, поэтому не внесли.

Евгения, прошу, не тратье мое время настолько непродуктивно...
я выдвинула вполне валидную теорию,почему данные не заполнены. настолько же валидную как и то, что вы предлагаете - человек не заполнил, пр..


#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* чтобы не исключать данные из последующего расчета искомого в рамках исследования, удаляя строки. лучше использовать медианное значение группы по критерию, определяюему гомогенную группу, в данном случае, для определения медианного уровня среднего дохода, можно использовать медианное значение типа занятости.
Медианое значение предпочтительней среднеарифметического, т.к. оно менее чувствительно к экстремальным выбросам в данных. 

Дополнила ответ.

### Шаг 4: общий вывод.

In [118]:
#дополнение:
all_factors = data.pivot_table(index = ['family_status','children','total_income_category','purpose_category'], values = 'debt', aggfunc = {'count','sum'})
all_factors = all_factors.rename(columns={'count':'no_debts','sum':'no_debt_overdue'})
all_factors['share_overdue']=100*round(all_factors['no_debt_overdue']/all_factors['no_debts'],4)
#расчет доли группы заемщиков от общего числа заемщиков
all_factors['%']=100*round(all_factors['no_debts']/q_purpose['no_debts'].sum(),4)
#отбираю не менее 50 случаев проосрочки в срезе, чтобы избежать экстремальных значений с небольшой частотой.
all_factors=all_factors[all_factors['no_debts']>=50].sort_values(by=['share_overdue','%'], ascending=[False,False])
all_factors.head(10)


no_debts  \
family_status         children total_income_category purpose_category                     
гражданский брак      1        C                     получение образования           73   
                                                     операции с автомобилем          76   
Не женат / не замужем 1        C                     операции с автомобилем          73   
гражданский брак      0        C                     получение образования          195   
                      1        C                     операции с недвижимостью       192   
женат / замужем       2        B                     получение образования           71   
Не женат / не замужем 0        C                     операции с автомобилем         396   
                      1        B                     операции с недвижимостью        66   
                      0        B                     операции с автомобилем         119   
женат / замужем       2        B                     операции с автомобилем          85   

                                                                               no_debt_overdue  \
family_status         children total_income_category purpose_category                            
гражданский брак      1        C                     получение образования                  14   
                                                     операции с автомобилем                 13   
Не женат / не замужем 1        C                     операции с автомобилем                 12   
гражданский брак      0        C                     получение образования                  29   
                      1        C                     операции с недвижимостью               26   
женат / замужем       2        B                     получение образования                   9   
Не женат / не замужем 0        C                     операции с автомобилем                 49   
                      1        B                     операции с недвижимостью                8   
                      0        B                     операции с автомобилем                 14   
женат / замужем       2        B                     операции с автомобилем                 10   

                                                                               share_overdue  \
family_status         children total_income_category purpose_category                          
гражданский брак      1        C                     получение образования             19.18   
                                                     операции с автомобилем            17.11   
Не женат / не замужем 1        C                     операции с автомобилем            16.44   
гражданский брак      0        C                     получение образования             14.87   
                      1        C                     операции с недвижимостью          13.54   
женат / замужем       2        B                     получение образования             12.68   
Не женат / не замужем 0        C                     операции с автомобилем            12.37   
                      1        B                     операции с недвижимостью          12.12   
                      0        B                     операции с автомобилем            11.76   
женат / замужем       2        B                     операции с автомобилем            11.76   

                                                                                  %  
family_status         children total_income_category purpose_category                
гражданский брак      1        C                     получение образования     0.34  
                                                     операции с автомобилем    0.36  
Не женат / не замужем 1        C                     операции с автомобилем    0.34  
гражданский брак      0        C                     получение образования     0.91  
                      1        C                     операции с недвижимостью  0.90  
женат / замужем       2        B                     п


Входные данные — статистика банка о платёжеспособности клиентов на основании 21.5 тыс. выданных ранее кредитов.

Основная задача исследования:влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. 

Доступные параметры:
*  children — количество детей в семье
*  days_employed — общий трудовой стаж в днях
*  dob_years — возраст клиента в годах
*  education — уровень образования клиента
*  education_id — идентификатор уровня образования
*  family_status — семейное положение
*  family_status_id — идентификатор семейного положения
*  gender — пол клиента
*  income_type — тип занятости
*  debt — имел ли задолженность по возврату кредитов
*  total_income — ежемесячный доход
*  purpose — цель получения кредита

Вывод: 
1. Выявлена корреляция между уровнем просрочки и количеством детей в семье:

*  заемщики без детей имеют самый низкий (7.5%) процент просрочки.
*  доля просрочек возрастает по мере роста числа детей: 1- 9.2%б 2- 9.7%.
*  тем не менее, семьи с 3 детьми допускают просрочку несколько реже -8.2%, чем с двумя. Предположительно, достигая                большего уровня достатка ко времени рождения третьего ребенка.
*  семьи с 4 детьми имеют наивысший уровень просрочки из всех сегментов выборки.

Число - Доля
детей   просрочек, %

4     - 9.9%

2     - 9.5%

1	  - 9.2%	

3	  - 8.2%

0	  - 7.5%


2. Выявлена, зависимость уровня просрочки от семейного положения заемщика:

*  Заемщики, не состоявшие в браке, имеют наибольший процент просрочки - 0.098%. Сраввнимые по уровню - группа заемщиков,          состоящих в гражданском браке. Предположительно, и те и другие - самая молодая страта в выборке, обладают меньшим уровнем      финансовой стабильности. 
*  Заемщики, состоящие в браке или в разводе незначительно лучше по уровню просрочки - 0.076% и 0.071% соответственно. 
*  Вдовцы демонстрируют самый низкий процент просрочки. Вероятно, в силу педантичности и стабильного достатка представителей      старшего поколения.

Также,исходя из исследования: наиболее вероятный профиль неблагонадежного заемщика - молодые люди (предположительно) со средним достатком от 50 до 200 т./мес. и ниже, неженатые или состоящие в гражданском браке, имеющие 1 или 2х детей, с целью кредита - покупка автомобиля или оплата образования. Сочетание всех перечисленных факторов в совокупности подробно не изучалось. Но каждый из перечисленных в отдельности, или в сочетании с другим, характеризует клиентов с повышенным риском просрочки.

Кредитором допущен довольно высокий уроверь просрочки платежей.
Предположительно, следует отдельно изучить уровень просрочки клиентов, не предоставивших данные о ежемесячном доходе и типе занятости. 
Организовать сбор данных (или восстановить при выгрузке) о доходе и стаже для таких заемщиков.
